### PR TITLE
Docs: Clean up of 'template' module docs

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This is part of a series of module doc cleanups.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
template

##### ANSIBLE VERSION
v2.8

### CODE REVIEW

File: lib/ansible/modules/files/template.py [hunk 1]
```diff
@@ -13,156 +17,152 @@
...
     type: bool
-    default: 'no'
+    default: no
     version_added: '2.6'
   force:
     description:
-      - the default is C(yes), which will replace the remote file when contents
-        are different than the source.  If C(no), the file will only be transferred
-        if the destination does not exist.
+    - Influence when the file is being transferred if the destination already exists.
```
Comment (acozine) [inline lib/ansible/modules/files/template.py]: I'd use "determine if the file is transferred" - "influence" sounds like the setting could be overridden by another paramreter.

File: lib/ansible/modules/files/template.py [hunk 2]
```diff
@@ -13,156 +17,152 @@
...
+    - When set to C(yes), replace the remote file when contents are different than the source.
+    - When set to C(no), the file will only be transferred if the destination does not exist.
     type: bool
-    default: 'yes'
+    default: yes
   follow:
     description:
-      - This flag indicates that filesystem links in the destination, if they exist, should be followed.
-      - Previous to Ansible 2.4, this was hardcoded as C(yes).
+    - Influence whether symbolic links should be followed.
```
Comment (acozine) [inline lib/ansible/modules/files/template.py]: "Determine" here too.

File: lib/ansible/modules/files/template.py [hunk 3]
```diff
@@ -13,156 +17,152 @@
...
-    i.e. C(#jinja2:variable_start_string:'[%', variable_end_string:'%]', trim_blocks: False)
-    which changes the variable interpolation markers to  [% var %] instead of  {{ var }}.
-    This is the best way to prevent evaluation of things that look like, but should not be Jinja2.
-    raw/endraw in Jinja2 will not work as you expect because templates in Ansible are recursively evaluated."
-  - You can use the C(copy) module with the C(content:) option if you prefer the template inline,
-    as part of the playbook.
+- Including a string that uses a date in the template will result in the template being marked 'changed' each time.
+- Since Ansible version 0.9, templates are loaded with C(trim_blocks=True).
+- >
+  Also, you can override jinja2 settings by adding a special header to template file.
```
Comment (acozine) [inline lib/ansible/modules/files/template.py:118]: Note to self: if this is not discussed elsewhere in the docs, we should add it.

File: lib/ansible/modules/files/template.py [hunk 4]
```diff
@@ -13,156 +17,152 @@
...
-    - files
-    - validate
+- files
+- validate
 '''
 
 EXAMPLES = r'''
-# Example from Ansible Playbooks
-- template:
+- name: Template a file to /etc/files.conf
```
Comment (acozine) [inline lib/ansible/modules/files/template.py:136]: +1 for adding `name:` lines

File: lib/ansible/utils/module_docs_fragments/files.py [hunk 5]
```diff
@@ -22,56 +8,64 @@ class ModuleDocFragment(object):
...
   unsafe_writes:
     description:
-      - By default this module uses atomic operations to prevent data
-        corruption or inconsistent reads from the target files,
-        but sometimes systems are configured or just broken in ways that prevent this. One example is docker mounted files,
-        which cannot be updated atomically from inside the container and can only be written in an unsafe manner.
-      - This option allows Ansible to fall back to unsafe methods of
-        updating files when atomic operations fail (however, it doesn't force Ansible to perform unsafe writes).
-        IMPORTANT! Unsafe writes are subject to race conditions and can lead to data corruption.
+    - Influence when to use atomic operation to prevent data corruption or inconsistent reads from the target file.
```
Comment (acozine) [inline lib/ansible/utils/module_docs_fragments/files.py:53]: "Determine" here too.

**Review by acozine — CHANGES_REQUESTED** [review]
Overall looks great - a few places where it could be even better. Thanks @dagwieers.

File: lib/ansible/modules/files/template.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/files/template.py]: Great, thanks !

File: lib/ansible/modules/files/template.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/files/template.py]: Fixed.

**Comment by dagwieers** [comment]
Please re-review !

**Review by acozine — APPROVED** [review]

**Pull request merged**